src/esphome/sensor/binary_sensor_map.cpp

```diff
@@ -0,0 +1,96 @@
+#include "esphome/defines.h"
+
+#ifdef USE_BINARY_SENSOR_MAP
+
+#include "esphome/sensor/binary_sensor_map.h"
+
+#include "esphome/log.h"
+#include <bitset>
+
+ESPHOME_NAMESPACE_BEGIN
+
+namespace sensor {
+
+static const char *TAG = "sensor.binary_sensor_map";
+
+BinarySensorMapChannel::BinarySensorMapChannel(binary_sensor::BinarySensor *sensor, float value) {
+  this->binary_sensor = sensor;
+  this->value = value;
+}
+
+BinarySensorMap::BinarySensorMap(const std::string &name) : Sensor(name) {}
+
+void BinarySensorMap::setup() { ESP_LOGCONFIG(TAG, "Setting up binary_sensor_map '%s'...", this->name_.c_str()); }
```

Comment:
If there's no setup logic we should remove this. The logs are only there for components that do some setup (and so it can quickly be found which component's setup failed)